File: mobile/lib/models/shopping_list_item.dart
class ShoppingListItem {
  final String id;
  final String householdId;
  final String ingredientId;
  final String? ingredientName;
  final double quantity;
  final String unit;
  final bool checked;
  final String source; // MANUAL / AUTO_GENERATED

  ShoppingListItem({
    required this.id,
    required this.householdId,
    required this.ingredientId,
    this.ingredientName,
    required this.quantity,
    required this.unit,
    required this.checked,
    required this.source,
  });

  factory ShoppingListItem.fromJson(Map<String, dynamic> json) => ShoppingListItem(
        id: json['id'] as String,
        householdId: json['householdId'] as String,
        ingredientId: json['ingredientId'] as String,
        ingredientName: json['ingredientName'] as String?,
        quantity: (json['quantity'] as num).toDouble(),
        unit: json['unit'] as String,
        checked: json['checked'] as bool,
        source: json['source'] as String,
      );
}

File: backend/src/main/java/com/dietscheduler/backend/shoppinglist/ShoppingListItem.java
package com.dietscheduler.backend.shoppinglist;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.util.UUID;

@Entity
@Table(name = "shopping_list_item")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ShoppingListItem {

    @Id
    @GeneratedValue
    private UUID id;

    @Column(name = "household_id", nullable = false)
    private UUID householdId;

    @Column(name = "ingredient_id", nullable = false)
    private UUID ingredientId;

    @Column(nullable = false, precision = 12, scale = 3)
    private BigDecimal quantity;

    @Column(nullable = false)
    private String unit;

    @Column(nullable = false)
    @Builder.Default
    private boolean checked = false;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ShoppingListSource source;
}
